docs/src/prototypes/CopyToClipboard/CopyToClipboard.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react'
+import { Popup, Button, Text, Portal, Ref } from '@stardust-ui/react/src'
+import * as copyToClipboard from 'copy-to-clipboard'
+
+export type CopyToClipboardProps = {
+  value: string
+  copyPrompt: string
+  timeout?: number
+  attached?: boolean
+  pointing?: boolean
+}
+
+class CopyToClipboard extends React.Component<CopyToClipboardProps> {
+  state = {
+    copied: false,
+    copiedHeight: 0,
+    copiedWidth: 0,
+  }
+  timeoutId = undefined
+  copiedTextRef = React.createRef<HTMLElement>()
+
+  static defaultProps = {
+    timeout: 4000,
+    attached: false,
+    pointing: false,
+  }
+
+  handleClick = () => {
+    if (this.timeoutId !== undefined) {
+      clearTimeout(this.timeoutId)
+    }
+    const { timeout, value } = this.props
+    this.setState({
+      copied: true,
+    })
+    this.timeoutId = setTimeout(() => {
+      this.setState({ copied: false })
+    }, timeout)
+    copyToClipboard(value)
+  }
+
+  componentDidMount() {
+    if (this.copiedTextRef.current === null) {
+      return
+    }
+    this.setState({
+      copiedHeight: this.copiedTextRef.current.clientHeight,
+      copiedWidth: this.copiedTextRef.current.clientWidth,
+    })
+  }
+
+  render() {
+    const { attached, pointing, copyPrompt } = this.props
+    const hideContent = this.state.copied && !attached
+    const copiedStr = 'Copied to clipboard'
+    const copiedText = hideContent ? (
```

Comment:
I am not fully understand this logic, can you please clarify?

I expect something like this:

```tsx
{attached && copied && <CopyButtonNotification>{notice}</CopyButtonNotification>}
{!attached && <Tooltip>{copied ? notice : usualMessage}</CopyButtonNotification>}
```

Reply:
We don't want to show the window-centered message when we are in _attached_ mode. In _attached_ mode, we just change the content of the Tooltip.

And to make the centered message more visible, I'm setting `Text`'s size to _larger.

Reply:
`hideContent` implies that we should hide the `Tooltip`'s content, as instead we are displaying the centered message.